ewc-affiliate/install-validator-centos-7-production-nethermind.sh

```diff
@@ -0,0 +1,493 @@
+
+echo "Account created: $ADDR"
+INFLUX_USER=${ADDR:2} # cutting 0x prefix
+INFLUX_PASS="$(openssl rand -hex 16)"
+
+
+# got the key now discard of the nethermind instance
+docker stop nethermind
+docker rm -f nethermind
+
+writeNethermindConfig
+NETHERMIND_KEY_FILE="$(ls -1 ./keystore/|grep UTC|tail -n1)"
+
+# Prepare nethermind telemetry pipe
+mkfifo /var/spool/nethermind.sock
+chown telegraf /var/spool/nethermind.sock
+
+# Write NLog config file
+wget $NLOG_CONFIG -O NLog.config
+setJsonRpcLogsLevelToError
+
+# Write the docker-compose file to disk
+writeDockerCompose
+
+# start everything up
+docker-compose up -d
+
+# Collect the enode from nethermind over RPC
+
+echo "Waiting 15 sec for nethermind to come up and generate the enode..."
+sleep 15
+ENODE=`curl -s --request POST --url http://localhost:8545/ --header 'content-type: application/json' --data '{ "method": "net_localEnode", "params": [], "id": 1, "jsonrpc": "2.0" }' | jq -r '.result'`
+
+# Now all information is complete to write the telegraf file
+writeTelegrafConfig
+service telegraf restart
+
+echo "Setting up firewall"
+
+systemctl enable iptables
+
+# non-docker services
+iptables -F INPUT
+iptables -F DOCKER-USER || true
+iptables -X FILTERS || true
+iptables -N FILTERS || true
+iptables -A FILTERS -p tcp --dport 2222 -j ACCEPT  # SSH
+iptables -A FILTERS -p tcp --dport 30303 -j ACCEPT # P2P tcp
+iptables -A FILTERS -p udp --dport 30303 -j ACCEPT # P2P udp
+iptables -A FILTERS -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT
+iptables -A FILTERS -j DROP
+
+# hook filters chain to input and docker
+iptables -A INPUT -p icmp --icmp-type any -j ACCEPT
+iptables -A INPUT -i lo -j ACCEPT
+iptables -A INPUT -j FILTERS
+iptables -P INPUT DROP
+
+iptables -A DOCKER-USER -i $NETIF -j FILTERS
+iptables -A DOCKER-USER -j RETURN
+service iptables save
+
+# run automated post-install audit
+cd /opt/
+wget https://downloads.cisofy.com/lynis/lynis-2.7.1.tar.gz
+tar xvzf lynis-2.7.1.tar.gz
+mv lynis /usr/local/
+ln -s /usr/local/lynis/lynis /usr/bin/lynis
+lynis audit system
+
+
+# Print install summary
+cd $HOMEDIR
+echo "==== EWF Affiliate Node Install Summary ====" > install-summary.txt
+echo "Company: $COMPANY_NAME" >> install-summary.txt
+echo "Validator Address: $ADDR" >> install-summary.txt
+echo "Enode: $ENODE" >> install-summary.txt
+echo "IP Address: $EXTERNAL_IP" >> install-summary.txt
+echo "InfluxDB Username: $INFLUX_USER" >> install-summary.txt
+echo "InfluxDB Password: $INFLUX_PASS" >> install-summary.txt
+cat install-summary.txt
+
+
+# END OF MAIN
+}
+
+## Files that get created
+
+writeDockerCompose() {
+cat > docker-compose.yml << 'EOF'
+version: '3.5'
+services:
+  nethermind:
+    image: ${NETHERMIND_VERSION}
+    restart: always
+    command:
+      --config ${CHAINNAME}
+    volumes:
+      - ./configs:/nethermind/configs:ro
+      - ./database:/nethermind/nethermind_db
+      - ./keystore:/nethermind/keystore
+      - ./logs:/nethermind/logs
```

Comment:
I dont like it :) 
If we wanna keep persistent log in files there should be a mechanism to rotate them, but still we keep duplicated logs streamed by docker containers with some rotation (lines ~404 daemon.json file).
Otherwise if we dont need them we should:
- remove bind mount for logs
- disable log writing inside docker container too

Reply:
btw. it is related to all installation files :)

Reply:
Ok, get it. Removed logs binding to keep it more simple.